sklearn/multioutput.py

```diff
@@ -636,22 +637,68 @@ def _check(self):
 
 class _BaseChain(BaseEstimator, metaclass=ABCMeta):
     _parameter_constraints: dict = {
-        "base_estimator": [HasMethods(["fit", "predict"])],
+        "base_estimator": [
+            HasMethods(["fit", "predict"]),
+            StrOptions({"deprecated"}),
+        ],
+        "estimator": [
+            HasMethods(["fit", "predict"]),
+            None,
+        ],
         "order": ["array-like", StrOptions({"random"}), None],
         "cv": ["cv_object", StrOptions({"prefit"})],
         "random_state": ["random_state"],
         "verbose": ["boolean"],
     }
 
+    # TODO(1.8): Remove base_estimator
     def __init__(
-        self, base_estimator, *, order=None, cv=None, random_state=None, verbose=False
+        self,
+        estimator=None,
+        *,
+        order=None,
+        cv=None,
+        random_state=None,
+        verbose=False,
+        base_estimator="deprecated",
     ):
+        self.estimator = estimator
         self.base_estimator = base_estimator
         self.order = order
         self.cv = cv
         self.random_state = random_state
         self.verbose = verbose
 
+    def _validate_estimator(self, default=None):
+        """Check the base estimator.
+
+        Set the `_estimator` attribute.
+        """
+        # TODO(1.8): Remove the self.base_estimator condition when
```

Comment:
This warning should be put on the top of the function because we will not need this function at all. All the code in the future will use `self.estimator` instead of `self._estimator` after the deprecation cycle.